### PR TITLE
Bug/2332 patterns in registrations

### PR DESCRIPTION
**Pull request by kzangeli**
Fixed issue #2332.
Also, when a ngsi9 registry, the duration is no longer returned (CNR?).

### CODE REVIEW

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -1,4 +1,5 @@
 - Fix: NGSIv2 subscriptions payload validation (#1964, #1965, #1967, #1973, #1974, #1975, #1979, #1980, #1981, #1983, #1984, #1988, #2007, #2018, #2093, #2095, #2099, #2100)
+- Fix: Return error on v1 register requests with entity isPattern set to TRUE (Issue #2332)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: Alternatve wording suggestion:

Fix: Return error on **NGSIv1** register requests with entity isPattern set to **"true"** (Issue #2332)

File: src/lib/serviceRoutines/postRegisterContext.cpp [hunk 2]
```diff
@@ -58,6 +58,28 @@ std::string postRegisterContext
...
+  //
+  for (unsigned int crIx = 0; crIx < rcrP->contextRegistrationVector.size(); ++crIx)
+  {
+    ContextRegistration* crP = rcrP->contextRegistrationVector[crIx];
+
+    for (unsigned int eIx = 0; eIx < crP->entityIdVector.size(); ++eIx)
+    {
+      if (isTrue(crP->entityIdVector[eIx]->isPattern))
+      {
+        std::string  details = "isPattern set to true for a registration";
```
Comment (fgalan) [inline src/lib/serviceRoutines/postRegisterContext.cpp]: I'd suggest:

"isPatter set to true for registrations is currently not supported"

to make clear that is not an error, but a kind of limitation at CB.

Comment (fgalan) [inline src/lib/serviceRoutines/postRegisterContext.cpp]: BTW, maybe the doc at https://fiware-orion.readthedocs.io/en/develop/user/context_providers/index.html should mention the limitation at the end in the "additional comments bullet list".

File: test/functionalTest/cases/2332_patterns_in_registrations/patterns_in_registrations.test [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+id patterns in registrations
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+
+--SHELL--
+
+#
+# 01. Try to register using isPattern=true, see error
```
Comment (fgalan) [inline test/functionalTest/cases/2332_patterns_in_registrations/patterns_in_registrations.test]: I'd suggest to add step 2 with an array of two registrations: one with isPattern false, the other with isPattern true. To check that the error-response logic also works in this case.

File: src/lib/serviceRoutines/postRegisterContext.cpp (hunk 2 above)
Comment (kzangeli) [inline src/lib/serviceRoutines/postRegisterContext.cpp]: Text fixed in 6782262
readthedocs still not fixed

File: CHANGES_NEXT_RELEASE (hunk 1 above)
Comment (kzangeli) [inline CHANGES_NEXT_RELEASE]: Fixed in 6782262

File: src/lib/serviceRoutines/postRegisterContext.cpp (hunk 2 above)
Comment (kzangeli) [inline src/lib/serviceRoutines/postRegisterContext.cpp]: readthedocs fixed in 8491415

File: test/functionalTest/cases/2332_patterns_in_registrations/patterns_in_registrations.test (hunk 3 above)
Comment (kzangeli) [inline test/functionalTest/cases/2332_patterns_in_registrations/patterns_in_registrations.test]: Fixed in 7406c24

**Comment by fgalan** [comment]
LGTM

**Pull request merged**